File: frontend/src/objects/system/app/planner/apiTypes/SaveDemoNoteRequest.ts
// Business note: planner protocol DTO shared with backend planner APIs; keep field names and value object types aligned.
import type { NoteBody, NoteStatus, NoteTitle } from '@/objects/core/SharedObjects'

export type SaveDemoNoteRequest = {
  title: NoteTitle
  body: NoteBody
  status: NoteStatus
}

File: backend/src/main/scala/system/app/planner/objects/SaveDemoNoteRequest.scala
package system.app.planner.objects

// Business note: planner-owned demo object; keep frontend planner mirrors aligned when this shape crosses the planner API boundary.
import system.objects.given

import system.objects.*

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class SaveDemoNoteRequest(
  title: NoteTitle,
  body: NoteBody,
  status: NoteStatus
)

object SaveDemoNoteRequest:
  given Encoder[SaveDemoNoteRequest] = deriveEncoder[SaveDemoNoteRequest]
  given Decoder[SaveDemoNoteRequest] = deriveDecoder[SaveDemoNoteRequest]
